From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mili <dev@bm.fun>
Date: Sat, 20 Jun 2026 00:00:10 +0800
Subject: [PATCH] Chunk Independent Scheduler - TickThread adaptation

Register CIS worker thread via ThreadLocal so TickThread.isTickThreadFor
recognises the worker as the owner of its assigned chunk.

diff --git a/ca/spottedleaf/moonrise/common/util/TickThread.java b/ca/spottedleaf/moonrise/common/util/TickThread.java
index 69cdd304d255d52c9b7dc9b6a33ffdb630b79abe..0000000000000000000000000000000000000000 100644
--- a/ca/spottedleaf/moonrise/common/util/TickThread.java
+++ b/ca/spottedleaf/moonrise/common/util/TickThread.java
@@ -169,4 +169,22 @@
     public static boolean isTickThreadFor(final Entity entity) {
         return isTickThread();
     }
+
+    // Mili start - CIS worker thread recognition
+    public static final ThreadLocal<Boolean> isCisWorker = ThreadLocal.withInitial(() -> false);
+    public static final ThreadLocal<Integer> cisWorkerChunkX = new ThreadLocal<>();
+    public static final ThreadLocal<Integer> cisWorkerChunkZ = new ThreadLocal<>();
+
+    public static void markCisWorker(int cx, int cz) {
+        isCisWorker.set(true);
+        cisWorkerChunkX.set(cx);
+        cisWorkerChunkZ.set(cz);
+    }
+
+    public static void unmarkCisWorker() {
+        isCisWorker.remove();
+        cisWorkerChunkX.remove();
+        cisWorkerChunkZ.remove();
+    }
+    // Mili end - CIS worker thread recognition
 }
